File: backend/internal/domain/notification.go
package domain

import "time"

// NotificationSettings represents user notification preferences
type NotificationSettings struct {
	UserID          int       `json:"user_id"`
	EmailEnabled    bool      `json:"email_enabled"`
	PushEnabled     bool      `json:"push_enabled"`
	NewPostsEnabled bool      `json:"new_posts_enabled"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

// Notification represents a notification sent to a user
type Notification struct {
	ID        int       `json:"id"`
	UserID    int       `json:"user_id" validate:"required"`
	Type      string    `json:"type" validate:"required"`
	Title     string    `json:"title" validate:"required,max=255"`
	Message   string    `json:"message" validate:"required"`
	Read      bool      `json:"read"`
	CreatedAt time.Time `json:"created_at"`
}

// NotificationType constants
const (
	NotificationTypeNewPost    = "new_post"
	NotificationTypeNewComment = "new_comment"
)

File: frontend/src/types/notification.ts
export const NotificationType = {
  NewPost: 'new_post',
  NewComment: 'new_comment',
  System: 'system',
} as const;

export type NotificationType = (typeof NotificationType)[keyof typeof NotificationType];

export interface Notification {
  id: number;
  user_id: number;
  type: NotificationType;
  title: string;
  message: string;
  read: boolean;
  created_at: string;
}

export interface NotificationSettings {
  user_id: number;
  email_enabled: boolean;
  push_enabled: boolean;
  new_posts_enabled: boolean;
}

export interface UpdateNotificationSettingsRequest {
  email_enabled?: boolean;
  push_enabled?: boolean;
  new_posts_enabled?: boolean;
}
